subsys/bluetooth/host/hci_core.c

```diff
@@ -5375,6 +5375,37 @@ int bt_le_scan_stop(void)
 }
 #endif /* CONFIG_BT_OBSERVER */
 
+int bt_le_set_chan_map(u8_t chan_map[5])
+{
+	struct bt_hci_cp_le_set_host_chan_classif *cp;
+	struct net_buf *buf;
+
+	if (!IS_ENABLED(CONFIG_BT_CENTRAL)) {
+		return -ENOTSUP;
+	}
+
+	if (!BT_CMD_TEST(bt_dev.supported_commands, 27, 3)) {
+		BT_WARN("Set Host Channel Classification command is "
+			"not supported");
+		return 0;
```

Comment:
I'm not sure we want to return success here. Wouldn't `-ENOTSUP` be equally appropriate here?

Reply:
I'm not sure, for set_privacy and set_flow_control we return success, so I followed that to be consistent.

Reply:
@joerchan with flow control I can understand the "silent fail" since we just fall back to not having it. It's also an internal detail that's not exposed to the application. With set_privacy() it looks like this is to prevent bt_id_add() from failing prematurely (code is from @MariuszSkamra so he can comment if he wants). In both cases these are internal details - the failure (or success) it not exposed to the application layer. In the case of this PR I do think it's fair to let the application know if the command succeeded or not.

Reply:
@jhedberg @joerchan This is done like this, because le_set_privacy_mode is only valid for controllers that support it. If the controller does not support the Privacy Mode setting i.e. network privacy, this command is omitted. So there is no need to report an error in this case.

Reply:
Alright, makes sense. Changed to ENOTSUP.